### ISSUE TITLE
Turkish language support

### ISSUE THREAD

**Issue by GreXLin85** [spencermountain/compromise#841]
Is there any Turkish language support or how can i implement it

**Comment by spencermountain** [comment]
hi @GreXLin85 no there isn't, but this would be a really cool (and big!) project.
I'd recommend starting with conjugating verbs in Turkish, which means you can start with a small dataset, and create a larger one with all the verb forms - that's a great first-step to creating something similar to compromise for turkish
cheers

**Comment by GreXLin85** [comment]
Is there any documents for how can i translate

**Comment by spencermountain** [comment]
no, i'm sorry - here's a document describing [how compromise works](https://observablehq.com/@spencermountain/compromise-internals), in english. That's the best I've got, i'm afraid.

**Comment by MarketingPip** [comment]
Here you go @GreXLin85 - this document I have created should tell you & other contributors how to translate this project into your own native language. 

---

**For Contributors**: How to translate Compromise.js for another language

To translate Compromise.js,  you can follow these generic steps:

1. Fork the Compromise.js repository on GitHub: Go to https://github.com/spencermountain/compromise and click on the "Fork" button in the top-right corner of the page. This will create a copy of the repository under your own GitHub account.

2. Clone the forked repository: On your local machine, open a terminal or command prompt and run the following command:

   ```
   git clone https://github.com/YOUR_USERNAME/compromise.git
   ```
   Replace `YOUR_USERNAME` with your actual GitHub username.

3. Install dependencies: Navigate into the cloned compromise directory using `cd compromise`, then install all required dependencies by running:
   ```
   npm install
   ```

4. Create a new branch for translation: It's recommended to work on translations in separate branches to keep changes organized and easy to manage. Create a new branch by executing:

    ```
   git checkout -b translation-branch-name
   ```
   Replace `translation-branch-name` with an appropriate name for your translation effort.

5. Translate files: The files you need to translate are located in `data` & `src`. These are various JSON & JavaScript files that define various linguistic terms used by Compromise.js.
   
    For example, if you want to translate Compromise.js into French (fr), navigate to `data`. There, you'll find multiple JSON files representing different aspects of language processing such as nouns, verbs, adjectives, etc.
    
    Edit these JSON files according to your target language's equivalents or translations for each term listed within them.

    There is also various terms stored inside scripts in `src` - which you will need to find & edit accordingly to your 
    language.

   You will also need to translate the Regex Rules: in the source folder you will also find Regex Rules that determine 
   various things (such as grammar rules & more) - which you will need also translate. 

   Note - you might need to modify the regex rules to apply the proper rules of grammar in your language.

6. Update documentation (optional): If you wish to add translated documentation or guides specific to your translated version of Compromise.js

7. Test your changes: To ensure everything is working as expected, modify the test files of Compromise.js for the language you translated to (or make your own tests).

9. Commit and push changes: Once you are satisfied with your translation work, commit your changes using Git:
  
   ```
    git add .
    git commit -m "Added translations for language X"
    git push origin translation-branch-name
   ```
  

10. Create a pull request: Go to the original Compromise.js repository on [GitHub](https://github.com/spencermountain/compromise)

---

> @spencermountain - feel free to steal / document / edit that for other contributors.
